src/zjs_promise.c

```diff
@@ -71,14 +54,10 @@ static jerry_value_t promise_resolve(const jerry_value_t function_obj,
         return zjs_error("promise handle not found");
     }
     if (fulfill) {
-        jerry_release_value(handle->then);
-        handle->then = jerry_acquire_value(argv[0]);
-        zjs_edit_js_func(handle->then_id, handle->then);
+        zjs_edit_js_func(handle->then_id, argv[0]);
         handle->then_set = 1;
```

Comment:
I don't think then_set and catch_set are being read anymore, only written, so remove them? Also, we could probably have always done this test as "then_id != -1" instead of having a separate boolean?